packages/k6/src/test-limits.ts

```diff
@@ -0,0 +1,141 @@
+import { check, JSONValue, sleep } from 'k6'
+import { SharedArray } from 'k6/data'
+import http from 'k6/http'
+import { Rate } from 'k6/metrics'
+import { Payload } from './config/types'
+import { vu } from 'k6/execution'
+
+// load the test duration from the environment or default to 1 hour
+let testDuration = '1h'
+if (__ENV.TEST_DURATION) {
+  testDuration = __ENV.TEST_DURATION
+}
+
+let RPS = 10
+if (__ENV.RPS && !isNaN(parseInt(__ENV.RPS))) {
+  RPS = parseInt(__ENV.RPS)
+}
+
+let T = 1 + 1 // Expect 1s response time + 1s network latency
+if (__ENV.T && !isNaN(parseInt(__ENV.T))) {
+  T = parseInt(__ENV.T)
+}
+
+let VU = RPS * T
+if (__ENV.VU && !isNaN(parseInt(__ENV.VU))) {
+  VU = parseInt(__ENV.VU)
+}
+
+export const errorRate = new Rate('errors')
+
+// load the test data
+const payloadPath = __ENV.PAYLOAD_PATH || '../src/config/http.json'
+const payloadData = new SharedArray('payloadData', () => {
+  return JSON.parse(open(payloadPath)) as Payload[]
+})
+
+function buildRequests() {
+  const requests = []
+  const params = {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  }
+
+  for (const payload of payloadData) {
+    requests.push({
+      body: payload.data,
+      params,
+    })
+  }
+
+  return requests
+}
+
+const requests = buildRequests()
+
+let uniqueRequests = Math.min(requests.length, VU)
+if (__ENV.UNIQUE_REQUESTS && parseInt(__ENV.UNIQUE_REQUESTS)) {
+  const envUniqueRequests = parseInt(__ENV.UNIQUE_REQUESTS)
+  if (envUniqueRequests > uniqueRequests) {
+    console.error(
+      `Requested to use ${envUniqueRequests} unique requests, but only have ${uniqueRequests}`,
+    )
+  } else if (envUniqueRequests > VU) {
+    console.error(`Requested to use ${envUniqueRequests} unqiue requests, but only have ${VU} VUs`)
+  } else {
+    uniqueRequests = envUniqueRequests
+  }
+}
+
+console.info(`Using ${uniqueRequests} unique requests`)
+
+// set the k6 running options
+export const options = {
+  thresholds: {
+    http_req_failed: ['rate<0.01'], // http errors should be less than 1%
+    http_req_duration: ['p(90)<1000'], // 90% of requests should be below 1s
+  },
+  stages: [
+    { duration: '5m', target: uniqueRequests }, // 5m warmup from 0 to `uniqueRequests`
+    { duration: '1m', target: VU }, // 1m warmup from `uniqueRequests` to target RPS
+    { duration: testDuration, target: VU }, // `testDuration` with constant target RPS
+  ],
```

Comment:
@boxhock This is what I had in mind as a stress test, the way the k6 docs show to do it is by setting many stages like
```suggestion
  stages: [
    { duration: '5m', target: uniqueRequests }, // 5m warmup from 0 to `uniqueRequests`
    { duration: testDuration / 10, target: (VU / 10) * 1 },
    { duration: testDuration / 10, target: (VU / 10) * 2 },
    { duration: testDuration / 10, target: (VU / 10) * 3 },
    { duration: testDuration / 10, target: (VU / 10) * 4 },
    { duration: testDuration / 10, target: (VU / 10) * 5 },
    { duration: testDuration / 10, target: (VU / 10) * 6 },
    { duration: testDuration / 10, target: (VU / 10) * 7 },
    { duration: testDuration / 10, target: (VU / 10) * 8 },
    { duration: testDuration / 10, target: (VU / 10) * 9 },
    { duration: testDuration / 10, target: VU },
  ],
```
We'd change the RPS variable to signify "max RPS", with the expectation that the test would fail somewhere along the line. Another option could be to use one of the [ramping executors](https://k6.io/docs/using-k6/scenarios/executors/ramping-vus/)

Reply:
I don't think we'd be able to make the test fail along the line at any point? With these stages, we'd just see the RPS increase gradually for each stage, and then see if it worked in the end or not afaict

Reply:
With the thresholds we'd be able to see when they fail, and optionally we can also abort execution as well https://k6.io/docs/using-k6/thresholds/#aborting-a-test-when-a-threshold-is-crossed

Reply:
@alejoberardino Thanks for this, I think I got it in a good state now, so please have a look. Only difference is that I added a scaleup period between each stage, allowing us to test for a constant RPS for some time instead of constantly scaling up. Let me know what you think